src/Umbraco.Web.UI.Client/src/common/directives/components/tree/umbtree.directive.js

```diff
@@ -243,33 +245,52 @@ function umbTreeDirective($q, $rootScope, treeService, notificationsService, use
                     if ($scope.customtreeparams) {
                         args["queryString"] = $scope.customtreeparams;
                     }
-
-                    return treeService.getTree(args)
-                        .then(function (data) {
-                            //Only use the tree data, if we are still on the correct section
-                            if(data.alias !== $scope.section){
-                                return $q.reject();
-                            }
-                            
-                            //set the data once we have it
-                            $scope.tree = data;
-
-                            //set the root as the current active tree
-                            $scope.activeTree = $scope.tree.root;
-
-                            emitEvent("treeLoaded", { tree: $scope.tree });
-                            emitEvent("treeNodeExpanded", { tree: $scope.tree, node: $scope.tree.root, children: $scope.tree.root.children });
-                         
-                            return $q.when(data);
-                        }, function (reason) {
-                            notificationsService.error("Tree Error", reason);
-                            return $q.reject(reason);
-                        });
+                    
+                    $scope.isLoading = true;
+                    $scope.isLoadingReason = "";
+                    
+                    return getTree(args);
                 }
                 else {
                     return $q.reject();
                 }
             }
+            
+            function getTree(args){
+                return treeService.getTree(args)
+                    .then(function (data) {
+
+
+                        //Only use the tree data, if we are still on the correct section
+                        if(data.alias !== $scope.section){
+                            return $q.reject();
+                        }
+
+                        //set the data once we have it
+                        $scope.tree = data;
+
+                        //set the root as the current active tree
+                        $scope.activeTree = $scope.tree.root;
+
+                        emitEvent("treeLoaded", { tree: $scope.tree });
+                        emitEvent("treeNodeExpanded", { tree: $scope.tree, node: $scope.tree.root, children: $scope.tree.root.children });
+
+                        $scope.isLoading = false;
+                        $scope.isLoadingReason = "";
+                        return $q.when(data);
+                    }, function (reason) {
+                        if(reason.status === 409){
```

Comment:
We can put this type of logic into angular interceptors, else we're going to have to duplicate this every single place in the application. We'll need an interceptor and then some custom service to display these messages. Will prob have to be some sort of new UX thing.

Reply:
Cool.. Sounds like a good solution